cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -2008,13 +2280,66 @@ class row_hasher {
    * @return A hash operator to use on the device
    */
   template <template <typename> class hash_function = cudf::hashing::detail::default_hash,
-            template <template <typename> class, typename>
+            template <template <typename> class, typename, template <typename> typename>
+            class DeviceRowHasher = device_row_hasher,
+            typename Nullate,
+            template <typename> typename dispatch_cond = type_identity_t>
+  DeviceRowHasher<hash_function, Nullate, dispatch_cond> device_hasher(
+    Nullate nullate = {}, uint32_t seed = DEFAULT_HASH_SEED) const
+  {
+    return DeviceRowHasher<hash_function, Nullate, dispatch_cond>(nullate, *d_t, seed);
+  }
+
+  /**
+   * @brief Get the hash operator to use on the device
+   *
+   * Returns a variant unary callable, `F`, with signature `hash_function::hash_value_type
+   * F(size_type)`. Concrete type in the returned variant is determined at runtime based on
+   * column_types.
+   *
+   * `F(i)` returns the hash of row i.
+   *
+   * @tparam Nullate A cudf::nullate type describing whether to check for nulls
```

Comment:
Seems like we might be missing documentation for the other template parameters here.